test/parallel/test-beforeexit-event.js

```diff
@@ -1,42 +1,29 @@
 'use strict';
-require('../common');
-var assert = require('assert');
-var net = require('net');
-var revivals = 0;
-var deaths = 0;
+const common = require('../common');
+const net = require('net');
 
-process.on('beforeExit', function() { deaths++; });
+const beforeExit = common.mustCall(() => {}, 4);
```

Comment:
Now that I think about it, we can actually get rid of this separate check entirely by just adding a

```js
process.on('beforeExit', common.mustCall(() => {}));
```

after the `this.close()` in `tryListen()`.